### ISSUE TITLE
Support for international typography

### ISSUE THREAD

**Issue by sujato** [material-components/material-components-web-components#1295]
Material design specifies some guidelines for international typography.

https://material.io/design/typography/language-support.html

However there doesn't seem to be any implementation in mwc, or upstream mdc for that matter.

What would international support look like?

### Line height

MD guidelines classify scripts into three types: 

1. English-like
2. Tall
3. Dense

It'd be lovely to have a simple mechanism for applying these throughout the design system. 

One close-to-the-platform way would be to apply these based on the language attribute, or more rigorously, the language + script attribute. An example from our code base:

```
  article[lang="ar"], section[lang="ar"],
  article[lang="fa"], section[lang="fa"] {
    line-height: 34px;
  }
```

### Right-to-left

Likewise the script type should trigger appropriate rtl display, ideally using CSS logical properties.

### Font styles

Certain font styles are only appropriate for English-like scripts and should generally be turned off for tall or dense scripts:

- Capitals and small-caps
- Italics
- Letter-spacing (might be appropriate for some, such as CJK, whereas not for say devanagari.)

The `font-synthesis` property is designed to prohibit creating faux italic and bold. It is nice in English-like scripts, but can be essential in some other scripts. In CJK for example, certain glyphs become unreadable if a faux bold is applied.

Note that these details are not part of the Material Design recommendations, but IMHO they should be! Also, `font-synthesis` is not in Chrome, which makes me sad.

## How to apply

A developer should ensure that the appropriate language tag, and if necessary, language + script tag, is applied to all elements containing text. Note that some languages, such as Sanskrit, can be expressed in multiple scripts. Here is a simplified example:

```
  [lang="my"],
 [lang="pli-Mymr"] {
    font-family: var(--myanmar-sans-font);
    font-size: var(--tall-font-size-md);
    line-height: var(--tall-font-line-height);
    font-variant-caps: normal;
    letter-spacing: normal;
    text-transform: none;
    font-synthesis: none;
  }
```

Once the language (and script) is identified, the appropriate changes should happen out of the box. The developer should not have to worry that text on a button is aligned properly if the user is reading Chinese.

A further refinement is to apply a line-height multiplier based on the font selection. If a site uses only Noto, then this is not needed. But if a non-standard font is used, the line-height will differ. To get around this, we measured the line-height of our basic font (Skolar) as compared to Noto, and found that Skolar * 1.13 gives us the Noto line-height. It would be nice to have a handy means of applying this. A developer should be able to simply set the line-height multiplier once.

Currently we implement these details by making long lists of language attributes and applying CSS to them. But it is complex and brittle and hard to test, as we support over 40 languages. 

Well, thanks for paying attention, and I hope something good can happen from this. Be happy!

**Comment by e111077** [comment]
Hello, thanks for the suggestions and bringing this up to our attention. For most of the internationalization guidance in the spec, we believe that most of these guidelines are app-level concerns, though there is some work that we believe can be done on our side e.g. i18n documentation, other hooks etc.

You can follow #1327 for overall i18n progress
